### PR TITLE
Fix some ball multiplier configs

### PR DESCRIPTION
**Pull request by Bassoonian**
Title

## **Discord contact info**
bassoonian

### CODE REVIEW

**Comment by kaicardenas2** [comment]
what does ball multiplier mean? is it like how it affects the catch rate?

**Comment by LOuroboros** [comment]
> what does ball multiplier mean? is it like how it affects the catch rate?

Ye. It's *the* modifier applied to a Poké Ball when you throw it in battle in order to catch a Pokémon.

Speaking of which, the modifier is still set to 150 for the Safari Ball even though the comment, @Bassoonian.

**Comment by kaicardenas2** [comment]
> Ye. It's _the_ modifier applied to a Poké Ball when you throw it in battle in order to catch a Pokémon.

make senses now that you put it that way 💀

**Review by LOuroboros — DISMISSED** [review]

File: include/config/battle.h [hunk 1]
```diff
@@ -159,10 +159,11 @@
 #define B_TIMER_BALL_MODIFIER       GEN_LATEST // In Gen5+, Timer Ball's effectiveness increases by x0.3 per turn instead of x0.1
 #define B_DUSK_BALL_MODIFIER        GEN_LATEST // In Gen7+, Dusk Ball's catch multiplier is x3 instead of x3.5.
 #define B_QUICK_BALL_MODIFIER       GEN_LATEST // In Gen5+, Quick Ball's catch multiplier is x5 instead of x4.
-#define B_LURE_BALL_MODIFIER        GEN_LATEST // In Gen7+, Lure Ball's catch multiplier is x5 instead of x3.
+#define B_LURE_BALL_MODIFIER        GEN_LATEST // In Gen7+, Lure Ball's catch multiplier is greater than x3.
```
Comment (AsparagusEduardo) [inline include/config/battle.h:162]:
```suggestion
#define B_LURE_BALL_MODIFIER        GEN_LATEST // In Gen8+, Lure Ball's catch multiplier is x4,. In Gen7, it's x5. In Gen6 and earlier, it's x3.
```

File: include/config/battle.h [hunk 2]
```diff
@@ -159,10 +159,11 @@
...
 #define B_HEAVY_BALL_MODIFIER       GEN_LATEST // In Gen7+, Heavy Ball's ranges change. See Cmd_handleballthrow.
 #define B_DREAM_BALL_MODIFIER       GEN_LATEST // In Gen8, Dream Ball's catch multiplier is x4 when the target is asleep or has the ability Comatose.
 #define B_SPORT_BALL_MODIFIER       GEN_LATEST // In Gen8, Sport Ball's catch multiplier was reduced from x1.5 to x1.
+#define B_SAFARI_BALL_MODIFIER      GEN_LATEST // In Gen8, Safari Ball's catch multiplier was reduced from x1.5 to x1.
```
Comment (AsparagusEduardo) [inline include/config/battle.h:166]:
```suggestion
#define B_DREAM_BALL_MODIFIER       GEN_LATEST // In Gen8+, Dream Ball's catch multiplier is x4 when the target is asleep or has the ability Comatose.
#define B_SPORT_BALL_MODIFIER       GEN_LATEST // In Gen8+, Sport Ball's catch multiplier was reduced from x1.5 to x1.
#define B_SAFARI_BALL_MODIFIER      GEN_LATEST // In Gen8+, Safari Ball's catch multiplier was reduced from x1.5 to x1.
```

File: src/battle_script_commands.c [hunk 3]
```diff
@@ -14901,7 +14905,7 @@ static void Cmd_handleballthrow(void)
                 break;
             case ITEM_LURE_BALL:
                 if (gIsFishingEncounter)
-                    ballMultiplier = (B_LURE_BALL_MODIFIER >= GEN_7 ? 500 : 300);
+                    ballMultiplier = (B_LURE_BALL_MODIFIER >= GEN_7 ? ((B_LURE_BALL_MODIFIER) >= GEN_8 ? 400: 500) : 300);
```
Comment (AsparagusEduardo) [inline src/battle_script_commands.c:14908]: This just looks ugly.
```suggestion
                {
                    if (B_LURE_BALL_MODIFIER >= GEN_8)
                        ballMultiplier = 400;
                    else if (B_LURE_BALL_MODIFIER >= GEN_7)
                        ballMultiplier = 500;
                    else
                        ballMultiplier = 300;
                }
```

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

File: src/battle_script_commands.c (hunk 3 above)
Comment (LOuroboros) [reply src/battle_script_commands.c:14908]: I don't agree, but I *would* say that newbies would have a harder time understanding it.

Comment (AlexOn1ine) [reply src/battle_script_commands.c]: We should decide whether we allow multi-ternaries or not.

Personally I don't mind them for configs but I dislike them in anything else.

Comment (LOuroboros) [reply src/battle_script_commands.c]: Afaiac, ternaries are fine for things like if/else, but when there's a 3rd statement involved it starts to get more and more convoluted the deeper you go into that rabbit hole, therefore it becomes less and less readable for newbies and eventually it just becomes too hard to interpret at a glance.

**Pull request merged**